nomenclate/core/nameparser.py

```diff
@@ -265,7 +270,8 @@ def get_string_camel_patterns(cls, name, min_length=0):
             # http://stackoverflow.com/questions/13954841/python-sort-upper-case-and-lower-case
             casing_permutations = list(set(cls._get_casing_permutations(abbr)))
             casing_permutations.sort(key=lambda v: (v.upper(), v[0].islower(), len(v)))
-            permutations = [permutation for permutation in casing_permutations if cls._valid_camel(permutation) or len(permutation) <= 2]
+            permutations = [permutation for permutation in casing_permutations if
+                            cls._valid_camel(permutation) or len(permutation) <= 2]
```

Comment:
the len is so we don't get any permutations under 2 characters which is an unsupported case.  I guess I can make that other method public, but I'll just rename to validate and return an error to follow the structure of the other error handling.